### PR TITLE
[asl] change behaviour of print + println

### PR DESCRIPTION
**Pull request by HadrienRenaud**
This PR introduces a sibling of `print`: `println`, that adds a new line at the end of the printed section. Now, `print` does not do that anymore.

It also remove the separators from between the different arguments to `print`/`println`.

For some examples, see `asllib/tests/print.t`.

### CODE REVIEW

**Comment by maranget** [comment]
Hi @HadrienRenaud. Test failures probably are easy to fix. Why not add one word or two to motivate the PR?

**Comment by maranget** [comment]
Also there are two `asl-630-behaviour-of-print` branches, this one et the other on origin. This is a bit confusing.

File: asllib/tests/print.t [hunk 1]
```diff
@@ -27,3 +25,36 @@
...
+  1234
+  0x03
+  TRUE
+  1/2
+  0
+  20
+
+  $ cat >print4.asl <<EOF
+  > func main () => integer begin
+  >   println ((1, 2));
```
Comment (Roman-Manevich) [inline asllib/tests/print.t:55]: Attempting to print tuples should result in a type error.

File: asllib/tests/print.t [hunk 2]
```diff
@@ -8,9 +8,7 @@
   > EOF
 
   $ aslref printer1.asl
-  Wow 2 3.14 some other string
-  no type-checking
-  32
+  Wow 2 157/50 some other stringno type-checking32
```
Comment (Roman-Manevich) [inline asllib/tests/print.t:11]: There should be no spaces between the individual sub-strings.

File: asllib/Interpreter.ml [hunk 3]
```diff
@@ -1036,9 +1037,10 @@ module Make (B : Backend.S) (C : Config) = struct
               vs
           else (
             List.map B.debug_value vs |> String.concat " " |> print_string;
```
Comment (Roman-Manevich) [inline asllib/Interpreter.ml:1039]: Remove the `" "` from `String.concat` - there should be no spaces between the individual strings.

File: asllib/error.ml [hunk 4]
```diff
@@ -168,6 +169,7 @@ let error_label = function
   | RecursionLimitReached -> "RecursionLimitReached"
   | EmptyConstraints -> "EmptyConstraints"
   | UnexpectedPendingConstrained -> "UnexpectedPendingConstrained"
+  | BadPrintType _ -> "BadPrintType"
```
Comment (Roman-Manevich) [inline asllib/error.ml:172]: Consider adding a type for a clearer type message. If there is a list of expressions, it's going to be hard to tell which one caused the problem.

File: asllib/tests/print.t (hunk 2 above)
Comment (Roman-Manevich) [reply asllib/tests/print.t:11]: Consider printing `0 as real`, `-1 as real`, `-1.0 as real`.

**Review by Roman-Manevich — APPROVED** [review]

**Comment by maranget** [comment]
Never mind. Having two branches with the same name lead to minor trouble with the `git switch` command. Nothing serious.

File: asllib/bitvector.mli [hunk 5]
```diff
@@ -81,6 +81,10 @@ val to_string : t -> string
 (** Returns a string representing the bitvector, indexed from the right and
     delimited by apostrophes. *)
 
+val to_string_hexa : t -> string
```
Comment (hrutvik) [inline asllib/bitvector.mli:84]: I feel like some information is lost here regarding the bit vector length. In particular:
- if the length is not a multiple of 4 you can't tell
- in the tests below it feels like there is an extra hex character, e.g. `'0111'` should be `0x7` but is `0x07`

I wonder if a solution is to print the new pseudo-syntax for bit vector literals of specific lengths, e.g. `0x07[:4]` or similar?

**Review by hrutvik — COMMENTED** [review]
Just a minor suggestion regarding printing of bit vectors, as I think it shouldn't lose length information

**Review by hrutvik — APPROVED** [review]

**Pull request merged**